include/args.h

```diff
@@ -205,31 +261,41 @@ struct ListObjectVersionsArgs : public ListObjectsCommonArgs {
 
   ListObjectVersionsArgs();
   ListObjectVersionsArgs(ListObjectsArgs args);
+  ~ListObjectVersionsArgs() = default;
 };  // struct ListObjectVersionsArgs
 
 struct PutObjectArgs : public PutObjectBaseArgs {
   std::istream &stream;
-  http::ProgressFunction progressfunc = NULL;
-  void *progress_userdata = NULL;
+  http::ProgressFunction progressfunc = nullptr;
+  void *progress_userdata = nullptr;
 
   PutObjectArgs(std::istream &stream, long object_size, long part_size);
-  error::Error Validate();
+  ~PutObjectArgs() = default;
+
+  error::Error Validate(); // PWTODO: the validator modifies the object
```

Comment:
If you don't like the TODO comments, can you suggest how we should track something that should be fixed?

Because I think TODO is better than nothing, especially if there is work to do.

Reply:
Suggest an alternative way of marking work to do.

Reply:
If these TODO means fixing issues not in this PR, please open Github issues. If these TODOs will be fixed in this PR in subsequent comments, make this PR as draft till the PR is completely ready.